src/user-event/clear.ts

```diff
@@ -0,0 +1,60 @@
+import { ReactTestInstance } from 'react-test-renderer';
+import { ErrorWithStack } from '../helpers/errors';
+import { isHostTextInput } from '../helpers/host-component-names';
+import { isPointerEventEnabled } from '../helpers/pointer-events';
+import { EventBuilder } from './event-builder';
+import { UserEventInstance } from './setup';
+import { dispatchEvent, wait } from './utils';
+import { emitTypingEvents } from './type/type';
+
+export async function clear(
+  this: UserEventInstance,
+  element: ReactTestInstance
+): Promise<void> {
+  if (!isHostTextInput(element)) {
+    throw new ErrorWithStack(
+      `clear() works only with host "TextInput" elements. Passed element has type "${element.type}".`,
+      clear
+    );
+  }
+
+  // Skip events if the element is disabled
+  if (element.props.editable === false || !isPointerEventEnabled(element)) {
```

Comment:
could we want to log or throw to inform the user the action is not possible? because I think if he did it, he did not realize his component is not editable and it would be nice to notify him of it

Reply:
I think this is a good idea to throw a helpful error in such case. 

There seems to be some discrepancy in OG User Event on that front:
* `clear()` and `click()`/`pointer()` seem to throw if element is not enabled
* `type()` seems to silently ignore the fact.

I think we should have consistent semantics in such case, i.e. always throw a helpful error. 

One thing to note, this would be again the Fire Event API which silently ignores such events.

@MattAgn @pierrezimmermannbam @AugustinLF @thymikee any thoughts on that more general subject?

Reply:
i agree, we should always throw errors and document it well so people are not too surprised it's different with FireEvent. And maybe we could also think if we want to change the behavior of fireEvent in the future

Reply:
Added throwing error here in case element is not editable nor pressable (pointer events prop).

For other types of elements, I'll open an issue for discussion.

Reply:
BTW `fireEvent` was already throwing errors in the past when no enable event handler was found but that has been removed. Here are refs to OG issue: https://github.com/callstack/react-native-testing-library/issues/469, and PR https://github.com/callstack/react-native-testing-library/pull/470